backend/builder/image_builder/image_builder.py

```diff
@@ -291,6 +292,7 @@ def _build_container(dockerfile_mount_path: str, image_tag: str) -> kubernetes.c
         args=args,
         volume_mounts=volume_mounts,
         security_context=container_security_context,
+        resources=get_resources_requirements(cpu_request="1000m", memory_request="4Gi", memory_limit="16Gi"),
```

Comment:
I'm afraid memory limit is too low:

<img width="1785" alt="image" src="https://github.com/Substra/substra-backend/assets/135698225/7fd77ed4-b9aa-41ff-9ed0-2c7cf8ef676a">

* First three small peaks: `torch` with cpu index
* Three last peaks: `torch` with GPU support 

---

```suggestion
        resources=get_resources_requirements(cpu_request="1000m", memory_request="4Gi", memory_limit="32Gi"),
```